src/main/java/de/metas/ui/web/handlingunits/HUEditorView.java

```diff
@@ -219,13 +219,19 @@ public HUEditorRow getById(final DocumentId rowId) throws EntityNotFoundExceptio
 	{
 		return rowsBuffer.getById(rowId);
 	}
-	
-	public HUEditorRow getParentRowByChildId(final DocumentId childId) throws EntityNotFoundException
+
+	public HUEditorRow getParentRowByChildIdOrNull(final DocumentId childId) throws EntityNotFoundException
 	{
-		return rowsBuffer.getParentRowByChildId(childId);
+		try
+		{
+			return rowsBuffer.getParentRowByChildId(childId);
+		}
+		catch (Exception e)
+		{
+			return null;
```

Comment:
why do we need to ignore the exception?
If you need to fetch parent only if found, consider having getParentRowByChildIdOrNull in rowsBuffer.